lib/resumption/server_exchange.go

```diff
@@ -156,25 +165,23 @@ func (r *SSHServerWrapper) handleResumptionExchangeV1(conn *multiplexer.Conn, dh
 	r.mu.Unlock()
 
 	if entry == nil {
-		r.log.Debug("Resumable connection not found or already deleted.")
-		_, _ = conn.Write([]byte{notFoundServerExchangeTag})
+		r.attemptHandover(conn, token)
 		return
 	}
 
-	if entry.remoteIP.IsValid() {
-		var remoteIP netip.Addr
-		if t, ok := conn.RemoteAddr().(*net.TCPAddr); ok {
-			remoteIP, _ = netip.AddrFromSlice(t.IP)
-		}
-		if entry.remoteIP != remoteIP.Unmap() {
-			r.log.Warn("Resumable connection attempted resumption from a different remote address.")
-			_, _ = conn.Write([]byte{badAddressServerExchangeTag})
-			return
-		}
-	} else {
-		r.log.Warn("Resumable connection attempted resumption on a connection with an invalid remote address.")
-		// the client can't recover from this, just send a "not found"
-		_, _ = conn.Write([]byte{notFoundServerExchangeTag})
+	var remoteIP netip.Addr
+	if t, _ := conn.RemoteAddr().(*net.TCPAddr); t != nil {
+		remoteIP, _ = netip.AddrFromSlice(t.IP)
+		remoteIP = remoteIP.Unmap()
+	}
+
+	r.resumeConnection(entry, conn, remoteIP)
+}
+
+func (r *SSHServerWrapper) resumeConnection(entry *connEntry, conn net.Conn, remoteIP netip.Addr) {
+	if entry.remoteIP != remoteIP {
+		r.log.Warn("Resumable connection attempted resumption from a different remote address.")
```

Comment:
I'm a bit reluctant to run it as an error because it can kind of happen naturally even if there's no bad actors involved, just as a result of changing network conditions.